src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -95,47 +108,78 @@ public void testCreateDataset() throws Exception {
         when(uriBuilder.replacePath(anyString())).thenReturn(uriBuilder);
         when(uriBuilder.build(anyString())).thenReturn(new URI("/api/dataset/1"));
         initResource();
-        Response response = resource.createDataset(authUser, uriInfo, new Gson().toJson(json));
+        Response response = resource.createDataset(authUser, uriInfo, json);
 
         assertEquals(201,response.getStatus());
     }
 
     @Test(expected = BadRequestException.class)
-    public void testCreateDatasetErrors() {
-        DataSet inUse = new DataSet();
-        when(datasetService.getDatasetByName("test")).thenReturn(inUse);
+    public void testCreateDatasetNoJson() {
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, "");
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetNoProperties() {
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, "{\"properties\":[]}");
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetMissingName() {
+        String json = createPropertiesJson("Dataset Name", null);
+        // TODO: Rewrite createDataset to handle empty name strings and entirely missing name properties
```

Comment:
This comment should be moved to two separate tests, something like `testCreateDatasetEmptyName` and `testCreateDatasetMissingName`. Each should test the class as it is and list a ticket/TODO to address in a subsequent PR. 

Additionally, this test should be renamed to `testCreateDatasetNullName` since it's not really a missing property, just null.